mlir/lib/Dialect/Linalg/Transforms/BlockPackMatmul.cpp

```diff
@@ -134,6 +136,24 @@ transposePackedMatmul(RewriterBase &rewriter, linalg::LinalgOp linalgOp,
   return packTransposedMatmul;
 }
 
+static SmallVector<int64_t> getDefaultBlockFactors(linalg::LinalgOp linalgOp) {
+  // get L1 cache size first.
+  uint32_t L1_cache_size = 4096; // default value
+  uint32_t cpuID = 0;
```

Comment:
nit: defaulting to device 0 is fine but I wouldn't assume it's a CPU

Reply:
I think it's just a naming issue, this should be `deviceID` or something. But also, I wouldn't assume the first one has it either. Perhaps a helper that, when not provided a device ID, returns a `pair<value, id>` if any device has it.

Reply:
Yes, will fix the naming issue.